pkg/operator/ceph/object/spec_test.go

```diff
@@ -155,3 +156,53 @@ func TestValidateSpec(t *testing.T) {
 	err = validateStore(context, s)
 	assert.Nil(t, err)
 }
+
+func testPodSpecPlacement(t *testing.T, hostNet bool, req int, placement *rook.Placement) {
+	c := newConfig()
+	c.clusterSpec = &cephv1.ClusterSpec{
+		Network: cephv1.NetworkSpec{HostNetwork: hostNet},
+	}
+
+	d := v1.PodSpec{
+		InitContainers:    []v1.Container{},
+		Containers:        []v1.Container{},
+		RestartPolicy:     v1.RestartPolicyAlways,
+		HostNetwork:       hostNet,
+		PriorityClassName: c.store.Spec.Gateway.PriorityClassName,
+	}
+
+	if placement != nil {
+		c.store.Spec.Gateway.Placement = *placement
+	}
+
+	c.setPodPlacement(&d, c.store.Spec.Gateway.Placement)
+
+	// should have a required anti-affnity
+	assert.Equal(t,
+		req,
+		len(d.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution))
+}
+
+func makePlacement() rook.Placement {
+	return rook.Placement{
+		PodAntiAffinity: &v1.PodAntiAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+				{
+					TopologyKey: v1.LabelZoneFailureDomain,
+				},
+			},
+		},
+	}
+}
+
+func TestPodSpecPlacement(t *testing.T) {
```

Comment:
I think I'd like to see that this is additionally tested at the `makeRGWPodSpec` level in `TestPodSpecs` at least once to verify that the `setPodPlacement` function is actually being called by the code. It should be sufficient to test that the anti-affinity we expect exists during the `hostNetworking: true` case there.